### PR TITLE
ci: improve test resilience and run style checks within `cargo nextest`

### PR DESCRIPTION
**Pull request by nagisa**
I got absolutely fed up with waiting for prerequisite infrastructure work for #9290 (why does it have to be _that_ hard?)

However the PR in question had some other important improvements that do not necessarily *rely* on changes to said infrastructure to work. In particular:

* `nextest` now retries failing tests a few times before giving up, to make sure they aren't spuriously failing;
  * This should help some timing out integration tests in particular, as those often fail because of some deadlockish situation in my experience.
* style checks still run with `cargo nextest` – unfortunately that means that in CI this will run these checks multiple times, but that doesn’t sound particularly terrible of a tradeoff (especially if the other changes mean we won't be retrying entire test suites as often anymore;)
* This should allow for a greater portion of the test suite to run on Macs – unfortunately not verified by the CI, but people do complain and this should make the situation better.

cc #9367

### CODE REVIEW

File: .config/nextest.toml [hunk 1]
```diff
@@ -4,3 +4,29 @@ slow-timeout = { period = "60s", terminate-after = 2, grace-period = "0s" }
...
+threads-required = 4
+
+[[profile.default.overrides]]
+filter = 'package(style-tests)'
+slow-timeout = { period = "120s", terminate-after = 5 }
+threads-required = 4
+
+# Unfortunately no support for inheriting profiles yet:
+# https://github.com/nextest-rs/nextest/issues/387
+[profile.ci]
```
Comment (nagisa) [inline .config/nextest.toml:16]: Argh! I can’t modify the CI scripts to enable this profile T_T

**Comment by nagisa** [comment]
@nikurt @Longarithm could I request a review here?

File: test-utils/style/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+fn rustfmt() {
+    let cargo = std::env::var_os("CARGO").unwrap_or(OsString::from("cargo"));
+    let mut cmd = Command::new(cargo);
+    cargo_env(&mut cmd);
+    cmd.args(&["fmt", "--", "--check"]);
+    ensure_success(cmd);
+}
+
+#[test]
+fn clippy() {
```
Comment (nikurt) [inline test-utils/style/src/lib.rs:49]: This function duplicates `scripts/run_clippy.sh`. I see that you plan to delete that file in #9290.
Should be fine, but maybe add a TODO to remove the duplicated files.

File: core/o11y/src/testonly.rs [hunk 3]
```diff
@@ -31,24 +31,24 @@ fn setup_subscriber_from_filter(mut env_filter: EnvFilter) {
 }
 
 pub fn init_test_logger() {
-    let env_filter = EnvFilter::new("tokio_reactor=info,tokio_core=info,hyper=info,debug");
+    let env_filter = EnvFilter::new("cranelift=warn,wasmtime=warn,h2=warn,tower=warn,trust_dns=warn,tokio_reactor=info,tokio_core=info,hyper=info,debug");
```
Comment (nikurt) [inline core/o11y/src/testonly.rs:34]: :+1:

File: core/o11y/src/io_tracer.rs [hunk 4]
```diff
@@ -117,7 +117,7 @@ impl<S: Subscriber + for<'span> LookupSpan<'span>> Layer<S> for IoTraceLayer {
         }
     }
 
-    #[allow(clippy::integer_arithmetic)]
+    #[allow(clippy::arithmetic_side_effects)]
```
Comment (nikurt) [inline core/o11y/src/io_tracer.rs:120]: Please also update `docs/practices/style.md`

```
285:`clippy::integer_arithmetic` lints.
```

**Review by nikurt — APPROVED** [review]

**Pull request merged**